adapters/adprime/adprime.go

```diff
@@ -71,19 +71,20 @@ func (a *AdprimeAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *ada
 		reqCopy.Imp[0].Ext = newExt
 
 		// keywords
-		if len(adprimeExt.Keywords) > 0 {
-			if reqCopy.Site == nil {
-				reqCopy.Site = &openrtb2.Site{}
-			}
-			reqCopy.Site.Keywords = strings.Join(adprimeExt.Keywords, ",")
+		if reqCopy.Site != nil && len(adprimeExt.Keywords) > 0 {
+			siteCopy := *reqCopy.Site
+			siteCopy.Keywords = strings.Join(adprimeExt.Keywords, ",")
+			reqCopy.Site = &siteCopy
```

Comment:
This is a slight change in logic when compared to f4166f4f, right? The f4166f4f version would copy the keywords found in `adprimeExt.Keywords` even if `reqCopy.Site` is `nil`. Are we sure we don't want to keep that logic and do something similar to the logic inside the `if len(adprimeExt.Audiences) > 0` in line 81?

Reply:
thanks for paying attention to this, we fixed the code as needed

Reply:
Do we want to check for `reqCopy.Site != nil` in both lines 74 and 81? Or did you mean to check for `reqCopy.User != nil` in line 81 instead?

On a second thought maybe my comment on my last review was misunderstood. I intention was to ask whether or not you'd want to keep initializing `reqCopy.Site` whenever there were elements inside `adprimeExt.Keywords` even if it was `nil`, just like before. I meant something like:
```
73         // keywords
74 -       if reqCopy.Site != nil && len(adprimeExt.Keywords) > 0 {
   +       if len(adprimeExt.Keywords) > 0 {
   +           if reqCopy.Site == nil {
   +               reqCopy.Site = &openrtb2.Site{}
   +           }
75             siteCopy := *reqCopy.Site
76             siteCopy.Keywords = strings.Join(adprimeExt.Keywords, ",")
77             reqCopy.Site = &siteCopy
78         }
79
80         // audiences
81 -       if reqCopy.Site != nil && len(adprimeExt.Audiences) > 0 {
   +       if len(adprimeExt.Audiences) > 0 {
82             if reqCopy.User == nil {
83                 reqCopy.User = &openrtb2.User{}
84             }
85             userCopy := *reqCopy.User
86             userCopy.CustomData = strings.Join(adprimeExt.Audiences, ",")
87             reqCopy.User = &userCopy
88         }
adapters/adprime/adprime.go
```
Sorry to be so inquisitive about this part of the code @Adprime

Reply:
Hi, `keywords` and `audiences` must be recorded in the `Site` and `User` objects, respectively, only on the condition that the "Site" object is in the bid request. If the `Site` object is present, but the `User` object is not present, then we will create it.

Reply:
I understand. Thank you for clarifying.